bin/fetch-proxy

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# If the first argument to this script is "latest" or unset, it fetches the
+# latest proxy binary via build.l5d.io/linkerd2-proxy/latest.txt. If it's set
+# to a 7-character commit sha from the master branch of the linkerd2-proxy repo,
+# it will fetch that binary instead.
+
+set -eu
+
+bindir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+rootdir="$( cd $bindir/.. && pwd )"
+builddir="$rootdir/target/proxy"
+
+version="${1:-latest}"
+latest_sha=""
+
+if [ "$version" == "latest" ]; then
+  out=$(curl -vsL https://build.l5d.io/linkerd2-proxy/latest.txt)
+  version=$(echo $out | awk '{print $2}' | sed 's/^linkerd2-proxy-//' | sed 's/\.tar\.gz//')
+  latest_sha=$(echo $out | awk '{print $1}')
+fi
+
+filename="linkerd2-proxy-${version}.tar.gz"
+archive="$builddir/$filename"
+
+mkdir -p "$builddir"
+curl -vsL "$builddir" "https://build.l5d.io/linkerd2-proxy/$filename" > "$archive"
+
+if [ -n "$latest_sha" ]; then
+  sha=$(sha256sum "$archive" | awk '{print $1}')
+  if [ "$sha" != "$latest_sha" ]; then
+    echo "sha mismatch" >&2
+    exit 1
+  fi
+fi
+
+tar -C "$builddir" -zxvf "$archive"
+mv "$builddir/linkerd2-proxy-$version/bin/linkerd2-proxy" "$builddir"
+echo "$version" > "$builddir/version.txt"
```

Comment:
What do you think about, instead of staging this into $builddir, it just fetches the tarball, validates sha, untars directory, removes tarball, and then _echoes the directory to stdout_.  Then, this script is generally useful as a tool that fetches the latest proxy and tells you what it is.  The part of this script that stages the directory release is pretty specific to the docker packaging scheme, imo

Reply:
Ah, actually, I figured out a way to echo the full path to the binary and also communicate the version sha. Let me know if it's consistent with what you were thinking.